docs/DeveloperGuide.md

```diff
@@ -68,7 +68,41 @@ add, delete and update`Expenditure`entries in a new`itemList`during testing with
 
 
 ## Implementation
-
+This section describes some noteworthy details on how certain features are implemented.
+
+### Add Command Feature
+#### Current Implementation
+The sequence diagram below shows the interactions of different subcomponents of the system when adding an expenditure
+to the list.  
+![add_command_sequence_diagram](images/Add_Command_Sequence_Diagram.png)  
+A key functionality of MindMyMoney is the ability to add and track user expenditure. A user can add in a new expenditure
+by specifying the payment type, the category, the description of the item, the cost of the item and the date and the 
+month purchased. 
+1. The `Parser` component parses user input and returns the new `AddCommand` object to the
+   `MindMyMoney`.
+2. `AddCommand` instantiates `addInput`, `expenditureList` and `creditCardList`.
+3. The application invokes `Addcommand#execute` to execute user instruction.
+4. During the execution, `Addcommand#execute` will parse through user input to obtain the `EXPENDITURE`, `CATEGORY`,
```

Comment:
Is there a reason why you used the '#' symbol instead of '.'? I think '.' would be better, unless it causes some formatting issues...
```suggestion
3. The application invokes `Addcommand.execute` to execute user instruction.
4. During the execution, `Addcommand.execute` will parse through user input to obtain the `EXPENDITURE`, `CATEGORY`,
```